### PR TITLE
feat: retrieve deployed contract

### PR DESCRIPTION
**Pull request by lgalabru**

### CODE REVIEW

File: addons/evm/src/commands/actions/call_contract.rs [hunk 1]
```diff
@@ -48,91 +48,106 @@ lazy_static! {
...
-                interpolable: true
+                tainting: true,
+                internal: false
             },
             contract_abi: {
                 documentation: "The contract ABI, optionally used to check input arguments before sending the transaction to the chain.",
                 typing: Type::addon(EVM_ADDRESS),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/call_contract.rs:79]: I'm not sure if this should be tainting - it is just used for type checking, to improve the error specificity. Chances are, if you change this, you'll be changing some _other_ field that _is_ tainting, but this one itself is not.

File: addons/evm/src/commands/actions/call_contract.rs [hunk 2]
```diff
@@ -48,91 +48,106 @@ lazy_static! {
...
+                internal: false
             },
             type: {
                 documentation: "The transaction type. Options are 'Legacy', 'EIP2930', 'EIP1559', 'EIP4844'. The default is 'EIP1559'.",
                 typing: Type::string(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/call_contract.rs:107]: I'm not sure the type should be tainting. This is more _how_ the tx call or deployment is made (specifically centered around gas pricing of the call) - if it's already been invoked/created with a given type I don't think the user cares about the type any more

File: addons/evm/src/commands/actions/call_contract.rs [hunk 3]
```diff
@@ -48,91 +48,106 @@ lazy_static! {
...
+                internal: false
             },
             gas_limit: {
                 documentation: "Sets the maximum amount of gas that should be used to execute this transaction.",
                 typing: Type::integer(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/call_contract.rs:142]: Don't think this is tainting

File: addons/evm/src/commands/actions/call_contract.rs [hunk 4]
```diff
@@ -48,91 +48,106 @@ lazy_static! {
...
+                internal: false
             },
             gas_price: {
                 documentation: "Sets the gas price for Legacy transactions.",
                 typing: Type::integer(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/call_contract.rs:149]: Don't think this is tainting

File: addons/evm/src/commands/actions/call_contract.rs (hunk 3 above)
Comment (MicaiahReid) [reply addons/evm/src/commands/actions/call_contract.rs:142]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/call_contract.rs (hunk 4 above)
Comment (MicaiahReid) [reply addons/evm/src/commands/actions/call_contract.rs:149]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/call_contract.rs (hunk 2 above)
Comment (MicaiahReid) [reply addons/evm/src/commands/actions/call_contract.rs:107]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/call_contract.rs (hunk 1 above)
Comment (MicaiahReid) [reply addons/evm/src/commands/actions/call_contract.rs:79]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/check_confirmations.rs [hunk 5]
```diff
@@ -25,28 +25,32 @@ lazy_static! {
...
                 },
                 rpc_api_url: {
-                  documentation: "The URL of the EVM API used to poll for the transaction's inclusion in a block.",
-                  typing: Type::string(),
-                  optional: false,
-                  interpolable: true
+                    documentation: "The URL of the EVM API used to poll for the transaction's inclusion in a block.",
+                    typing: Type::string(),
+                    optional: false,
+                    tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/check_confirmations.rs:38]: It's interesting that the api url isn't tainting for call_contract, but it is for this - is that on purpose?

File: addons/evm/src/commands/actions/deploy_contract.rs [hunk 6]
```diff
@@ -47,97 +47,113 @@ lazy_static! {
...
-                interpolable: true
+                tainting: true,
+                internal: false
             },
             type: {
                 documentation: "The transaction type. Options are 'Legacy', 'EIP2930', 'EIP1559', 'EIP4844'. The default is 'EIP1559'.",
                 typing: Type::string(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/deploy_contract.rs:78]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/deploy_contract.rs [hunk 7]
```diff
@@ -47,97 +47,113 @@ lazy_static! {
...
+                internal: false
             },
             max_fee_per_gas: {
                 documentation: "Sets the max fee per gas of an EIP1559 transaction.",
                 typing: Type::integer(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/deploy_contract.rs:85]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/deploy_contract.rs [hunk 8]
```diff
@@ -47,97 +47,113 @@ lazy_static! {
...
+                internal: false
             },
             max_priority_fee_per_gas: {
                 documentation: "Sets the max priority fee per gas of an EIP1559 transaction.",
                 typing: Type::integer(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/deploy_contract.rs:92]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/deploy_contract.rs [hunk 9]
```diff
@@ -47,97 +47,113 @@ lazy_static! {
...
+                internal: false
             },
             gas_limit: {
                 documentation: "Sets the maximum amount of gas that should be used to execute this transaction.",
                 typing: Type::integer(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/deploy_contract.rs:113]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/deploy_contract.rs [hunk 10]
```diff
@@ -47,97 +47,113 @@ lazy_static! {
...
+                internal: false
             },
             gas_price: {
                 documentation: "Sets the gas price for Legacy transactions.",
                 typing: Type::integer(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/deploy_contract.rs:120]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/deploy_contract.rs [hunk 11]
```diff
@@ -47,97 +47,113 @@ lazy_static! {
...
+                internal: false
             },
             confirmations: {
                 documentation: "Once the transaction is included on a block, the number of blocks to await before the transaction is considered successful and Runbook execution continues.",
                 typing: Type::integer(),
                 optional: true,
-                interpolable: true
+                tainting: false,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/deploy_contract.rs:141]: Confirmations is tainting on the check_confirmations function, but not here. Is that right?

File: addons/evm/src/commands/actions/deploy_contract_create2.rs [hunk 12]
```diff
@@ -50,133 +50,155 @@ lazy_static! {
...
-                interpolable: true
+                tainting: true,
+                internal: false
             },
             type: {
                 documentation: "The transaction type. Options are 'Legacy', 'EIP2930', 'EIP1559', 'EIP4844'. The default is 'EIP1559'.",
                 typing: Type::string(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/deploy_contract_create2.rs:109]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/deploy_contract_create2.rs [hunk 13]
```diff
@@ -50,133 +50,155 @@ lazy_static! {
...
+                internal: false
             },
             max_fee_per_gas: {
                 documentation: "Sets the max fee per gas of an EIP1559 transaction.",
                 typing: Type::integer(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/deploy_contract_create2.rs:116]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/deploy_contract_create2.rs [hunk 14]
```diff
@@ -50,133 +50,155 @@ lazy_static! {
...
+                internal: false
             },
             max_priority_fee_per_gas: {
                 documentation: "Sets the max priority fee per gas of an EIP1559 transaction.",
                 typing: Type::integer(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/deploy_contract_create2.rs:123]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/deploy_contract_create2.rs [hunk 15]
```diff
@@ -50,133 +50,155 @@ lazy_static! {
...
+                internal: false
             },
             gas_limit: {
                 documentation: "Sets the maximum amount of gas that should be used to execute this transaction.",
                 typing: Type::integer(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/deploy_contract_create2.rs:144]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/deploy_contract_create2.rs [hunk 16]
```diff
@@ -50,133 +50,155 @@ lazy_static! {
...
+                internal: false
             },
             gas_price: {
                 documentation: "Sets the gas price for Legacy transactions.",
                 typing: Type::integer(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/deploy_contract_create2.rs:151]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/eth_call.rs [hunk 17]
```diff
@@ -30,91 +30,106 @@ lazy_static! {
...
-                interpolable: true
+                tainting: true,
+                internal: false
             },
             type: {
                 documentation: "The transaction type. Options are 'Legacy', 'EIP2930', 'EIP1559', 'EIP4844'. The default is 'EIP1559'.",
                 typing: Type::string(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/eth_call.rs:89]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/eth_call.rs [hunk 18]
```diff
@@ -30,91 +30,106 @@ lazy_static! {
...
+                internal: false
             },
             max_fee_per_gas: {
                 documentation: "Sets the max fee per gas of an EIP1559 transaction.",
                 typing: Type::integer(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/eth_call.rs:96]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/eth_call.rs [hunk 19]
```diff
@@ -30,91 +30,106 @@ lazy_static! {
...
+                internal: false
             },
             max_priority_fee_per_gas: {
                 documentation: "Sets the max priority fee per gas of an EIP1559 transaction.",
                 typing: Type::integer(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/eth_call.rs:103]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/eth_call.rs [hunk 20]
```diff
@@ -30,91 +30,106 @@ lazy_static! {
...
+                internal: false
             },
             gas_limit: {
                 documentation: "Sets the maximum amount of gas that should be used to execute this transaction.",
                 typing: Type::integer(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/eth_call.rs:124]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/eth_call.rs [hunk 21]
```diff
@@ -30,91 +30,106 @@ lazy_static! {
...
+                internal: false
             },
             gas_price: {
                 documentation: "Sets the gas price for Legacy transactions.",
                 typing: Type::integer(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/eth_call.rs:131]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/sign_transfer.rs [hunk 22]
```diff
@@ -42,79 +42,85 @@ lazy_static! {
...
-                interpolable: true
+                tainting: true,
+                internal: false
             },
             type: {
                 documentation: "The transaction type. Options are 'Legacy', 'EIP2930', 'EIP1559', 'EIP4844'. The default is 'EIP1559'.",
                 typing: Type::string(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/sign_transfer.rs:73]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/sign_transfer.rs [hunk 23]
```diff
@@ -42,79 +42,85 @@ lazy_static! {
...
+                internal: false
             },
             max_fee_per_gas: {
                 documentation: "Sets the max fee per gas of an EIP1559 transaction.",
                 typing: Type::integer(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/sign_transfer.rs:80]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/sign_transfer.rs [hunk 24]
```diff
@@ -42,79 +42,85 @@ lazy_static! {
...
+                internal: false
             },
             max_priority_fee_per_gas: {
                 documentation: "Sets the max priority fee per gas of an EIP1559 transaction.",
                 typing: Type::integer(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/sign_transfer.rs:87]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/sign_transfer.rs [hunk 25]
```diff
@@ -42,79 +42,85 @@ lazy_static! {
...
+                internal: false
             },
             gas_limit: {
                 documentation: "Sets the maximum amount of gas that should be used to execute this transaction.",
                 typing: Type::integer(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/sign_transfer.rs:108]:
```suggestion
                tainting: false,
```

File: addons/evm/src/commands/actions/sign_transfer.rs [hunk 26]
```diff
@@ -42,79 +42,85 @@ lazy_static! {
...
+                internal: false
             },
             gas_price: {
                 documentation: "Sets the gas price for Legacy transactions.",
                 typing: Type::integer(),
                 optional: true,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/evm/src/commands/actions/sign_transfer.rs:115]:
```suggestion
                tainting: false,
```

File: addons/sp1/src/commands/actions/create_proof.rs [hunk 27]
```diff
@@ -24,25 +24,29 @@ lazy_static! {
...
-                    interpolable: true
+                    tainting: true,
+                    internal: false
                 },
                 sp1_private_key: {
                     documentation: "Verify proof locally.",
                     typing: Type::string(),
                     optional: true,
-                    interpolable: true
+                    tainting: true,
```
Comment (MicaiahReid) [inline addons/sp1/src/commands/actions/create_proof.rs:48]:
```suggestion
                    tainting: false,
```
If I've already verified successfully locally, then decide on the next one I want to use a key, I don't think setting it should rerun

File: addons/stacks/src/commands/actions/broadcast_transaction.rs [hunk 28]
```diff
@@ -32,38 +32,44 @@ lazy_static! {
...
-                    interpolable: true
+                    tainting: true,
+                    internal: false
                 },
                 rpc_api_url: {
                     documentation: "The URL to use when making API requests.",
                     typing: Type::string(),
                     optional: true,
-                    interpolable: true
+                    tainting: true,
```
Comment (MicaiahReid) [inline addons/stacks/src/commands/actions/broadcast_transaction.rs:49]: I thought in others the rpc url was false?

Comment (MicaiahReid) [reply addons/stacks/src/commands/actions/broadcast_transaction.rs:49]:
```suggestion
                    tainting: false,
```

File: crates/txtx-addon-kit/src/types/signers.rs [hunk 29]
```diff
@@ -292,6 +292,18 @@ impl SignerInstance {
         }
     }
 
+    pub fn compute_fingerprint(&self, evaluated_inputs: &CommandInputsEvaluationResult) -> Did {
+        let mut comps = vec![];
+        for input in self.specification.inputs.iter() {
+            let Some(value) = evaluated_inputs.inputs.get_value(&input.name) else { continue };
+            if input.sensitive {
+                println!("Including {} for fingerprint", input.name);
```
Comment (MicaiahReid) [inline crates/txtx-addon-kit/src/types/signers.rs:300]:
```suggestion
```

File: addons/stacks/src/commands/actions/send_stx.rs [hunk 30]
```diff
@@ -40,55 +40,64 @@ lazy_static! {
...
-                    interpolable: true
+                    tainting: true,
+                    internal: false
                     },
                 rpc_api_url: {
                     documentation: "The URL to use when making API requests.",
                     typing: Type::string(),
                     optional: true,
-                    interpolable: true
+                    tainting: true,
```
Comment (MicaiahReid) [inline addons/stacks/src/commands/actions/send_stx.rs:64]:
```suggestion
                    tainting: false,
```

File: addons/stacks/src/commands/actions/send_stx.rs [hunk 31]
```diff
@@ -40,55 +40,64 @@ lazy_static! {
...
+                    internal: false
                 },
                 rpc_api_auth_token: {
                     documentation: "The HTTP authentication token to include in the headers when making API requests.",
                     typing: Type::string(),
                     optional: true,
-                    interpolable: true
+                    tainting: true,
```
Comment (MicaiahReid) [inline addons/stacks/src/commands/actions/send_stx.rs:71]:
```suggestion
                    tainting: false,
```

File: addons/stacks/src/commands/actions/send_stx.rs [hunk 32]
```diff
@@ -40,55 +40,64 @@ lazy_static! {
...
+                    internal: false
                 },
                 fee: {
                     documentation: "The transaction fee. This value will automatically be estimated if omitted.",
                     typing: Type::integer(),
                     optional: true,
-                    interpolable: true
+                    tainting: true,
```
Comment (MicaiahReid) [inline addons/stacks/src/commands/actions/send_stx.rs:99]:
```suggestion
                    tainting: false,
```

File: addons/stacks/src/commands/actions/deploy_contract.rs [hunk 33]
```diff
@@ -148,48 +165,56 @@ lazy_static! {
...
-                    interpolable: true
+                    tainting: true,
+                    internal: false
                 },
                 fee_strategy: {
                     documentation: "The strategy to use for automatically estimating fee ('low', 'medium', 'high'). Default to 'medium'.",
                     typing: Type::string(),
                     optional: true,
-                    interpolable: true
+                    tainting: true,
```
Comment (MicaiahReid) [inline addons/stacks/src/commands/actions/deploy_contract.rs:209]:
```suggestion
                    tainting: false,
```

File: addons/stacks/src/commands/actions/deploy_contract.rs [hunk 34]
```diff
@@ -148,48 +165,56 @@ lazy_static! {
...
                 },
                 contracts_ids_dependencies: {
                     documentation: "Contracts that are depending on this contract at their deployment.",
                     typing: Type::array(Type::string()),
                     optional: true,
-                    interpolable: true
+                    tainting: true,
+                    internal: false
                 },
                 contracts_ids_lazy_dependencies: {
```
Comment (MicaiahReid) [inline addons/stacks/src/commands/actions/deploy_contract.rs:198]: I think we need a better name for this field and the above `contracts_ids_dependencies`. It's weird that all three words are plural.

`dependency_contract_ids` & `lazy_dependency_contract_ids`? Is "lazy dependency" a known concept? I've never heard of that before

File: addons/stacks/src/commands/actions/deploy_contract.rs [hunk 35]
```diff
@@ -57,88 +60,102 @@ lazy_static! {
...
-                    interpolable: true
+                    tainting: true,
+                    internal: false
                 },
                 fee: {
                     documentation: "The transaction fee. This value will automatically be estimated if omitted.",
                     typing: Type::integer(),
                     optional: true,
-                    interpolable: true
+                    tainting: true,
```
Comment (MicaiahReid) [inline addons/stacks/src/commands/actions/deploy_contract.rs:143]:
```suggestion
                    tainting: false,
```

File: addons/stacks/src/commands/actions/sign_transaction.rs [hunk 36]
```diff
@@ -53,43 +53,50 @@ lazy_static! {
...
-                interpolable: true
+                tainting: true,
+                internal: false
             },
             fee: {
                 documentation: "The transaction fee. This value will automatically be estimated if omitted.",
                 typing: Type::integer(),
                 optional: false,
-                interpolable: true
+                tainting: true,
```
Comment (MicaiahReid) [inline addons/stacks/src/commands/actions/sign_transaction.rs:91]:
```suggestion
                tainting: false,
```

File: addons/telegram/src/send_message.rs [hunk 37]
```diff
@@ -22,25 +22,29 @@ lazy_static! {
...
-                interpolable: true
+                tainting: false,
+                internal: false
             },
             telegram_bot_api_token: {
               documentation: "Coming soon",
               typing: Type::string(),
               optional: false,
-              interpolable: true
+              tainting: true,
```
Comment (MicaiahReid) [inline addons/telegram/src/send_message.rs:32]:
```suggestion
              tainting: false,
```

File: addons/telegram/src/setup_chat.rs [hunk 38]
```diff
@@ -22,13 +22,15 @@ lazy_static! {
...
-                interpolable: true
+                tainting: false,
+                internal: false
             },
             telegram_bot_api_token: {
               documentation: "Coming soon",
               typing: Type::string(),
               optional: false,
-              interpolable: true
+              tainting: true,
```
Comment (MicaiahReid) [inline addons/telegram/src/setup_chat.rs:32]:
```suggestion
              tainting: false,
```

**Review by MicaiahReid — CHANGES_REQUESTED** [review]
Sorry this is a lot 😅

Also, what's the `internal` flag for?

**Comment by MicaiahReid** [comment]
Any thoughts on making `tainting`/`internal` optional fields for the macro with a default value? It would be nice when making these, but I admit you run the risk of forgetting to set the field.

**Comment by MicaiahReid** [comment]
One last complaint: I really don't love the word "tainting" for this. It feels like internal kitchen stuff that driver authors shouldn't need to be aware of? Would something like `critical` work?

File: addons/evm/src/commands/actions/check_confirmations.rs (hunk 5 above)
Comment (lgalabru) [reply addons/evm/src/commands/actions/check_confirmations.rs:38]: Addressed with latest

**Comment by lgalabru** [comment]
@MicaiahReid

> what's the internal flag for?

Internal stands for inputs needed internally, that we don't want to expose in the documentation

> dependency_contract_ids & lazy_dependency_contract_ids

Addressed with latest commit

> Is "lazy dependency" a known concept?

In this context, it means that the dependencies can be in the same block - vs committed in a prior block.

> It's interesting that the api url isn't tainting for call_contract, but it is for this

Addressed with latest commit
 
> It feels like internal kitchen stuff that driver authors shouldn't need to be aware of?

Unfortunately this work needs to be done manually, and a default value would lead to bugs. I don't see how we can infer this, but could be missing something 

>  I really don't love the word "tainting" for this

The tainting concept / terminology is coming from Terraform where tainted resources are detected automatically but can also be done manually through a command line - `txtx taint <runbook> <action>` for forcing the re-execution of a subgraph. Open to another idea, but I don't see `critical` as an adequate alternative.

**Comment by MicaiahReid** [comment]
> Internal stands for inputs needed internally, that we don't want to expose in the documentation

Very nice 💯 

>The tainting concept / terminology is coming from Terraform where tainted resources are detected automatically but can also be done manually through a command line - `txtx taint <runbook> <action>` for forcing the re-execution of a subgraph. Open to another idea, but I don't see `critical` as an adequate alternative.

Okay, understood. How about `can_taint_execution` or `can_taint_state`? The `ing` suffix feels off, I'd love to have it worded more like a bool. After that, I'm good!

**Comment by lgalabru** [comment]
We're currently using adjectives for qualifying inputs (`internal` vs `is_internal` , `optional` vs `is_optional`, etc). 
Can you think of an alternative adjective for qualifying these `tainting` inputs?

**Comment by MicaiahReid** [comment]
Sure, but I'd argue that a bool like `internal`/`optional` is much clearer in this case - they say what this input _is_, while tainting describes what it _does_. 

Since as you said tainting is a specific concept with Terraform, I don't think we should look for another adjective, but rather word it in a way that is less awkward and makes the purpose of the field clearer.

**Review by MicaiahReid — APPROVED** [review]

**Pull request merged**